src/main/kotlin/dev/architectury/plugin/ArchitecturyPluginExtension.kt

```diff
@@ -480,6 +484,77 @@ open class ArchitectPluginExtension(val project: Project) {
             action(it)
         }
     }
+
+    /**
+     * Inherits this loader-specific project from a common project.
+     * This function also sets up bundling by default, which can be configured with the [action].
+     *
+     * The value of [commonProject] can be a [Project] object or a [CharSequence] of its path.
+     *
+     * This function must be called after the Architectury Plugin mod loader has been set.
+     */
+    @JvmOverloads
+    fun fromCommon(commonProject: Any?, action: Action<FromCommonSettings> = EmptyAction.cast()) {
+        val loader = this.loader
+            ?: throw IllegalStateException("architectury.fromCommon must be called after the loader has been set")
+
+        // Resolve the project
+        val (commonPath, common) = when (commonProject) {
+            is Project -> Pair(commonProject.path, commonProject)
+            is CharSequence -> {
+                val path = commonProject.toString()
+                Pair(path, project.project(path))
+            }
+            else -> throw IllegalArgumentException("Cannot convert value '$commonProject' to a Gradle project")
+        }
+
+        // Find the production jar from the common project.
+        val productionCommonJar = common.tasks.named(
+            "transformProduction${loader.titledId}",
+            TransformingTask::class.java
+        )
+
+        // Create settings object, initialise it with defaults and pass it through the action.
+        val settings = project.objects.newInstance(FromCommonSettings::class.java)
+        with(settings) {
+            val defaultBundle = project.objects.newInstance(FromCommonSettings.BundleSettings::class.java)
+            bundle.convention(defaultBundle)
+            defaultBundle.enabled.convention(true)
+            action.execute(this)
+        }
+
+        // Add compile + runtime dependency for the common project.
+        val commonDevDep = project.dependencies.project(
+            mapOf(
+                "path" to commonPath,
+                "configuration" to "namedElements"
+            )
+        )
+        if (commonDevDep is ModuleDependency) {
+            commonDevDep.isTransitive = false
+        }
+        project.dependencies.add(JavaPlugin.IMPLEMENTATION_CONFIGURATION_NAME, commonDevDep)
```

Comment:
TODO: Check how this publishes. Using `namedElements` over the default config should prevent it from appearing in the POM though